pages/linux/eu-readelf.md

```diff
@@ -0,0 +1,8 @@
+# eu-readelf
+
+> Displays information about ELF files.
+> More information: <https://manned.org/eu-readelf>.
+
+- Displays the contents of the NOTE segments and sections of a file:
+
+`eu-readelf -n {{path/to/file}}`
```

Comment:
```suggestion
`eu-readelf {{-n|--notes}} {{path/to/file}}`
```
Is the file placeholder correct in this context?

The manned page refers to it as:

```md
-n
       --notes [section name]
           Displays the contents of the NOTE segments and/or sections, if any.
```

Reply:
It seems that `[section name]` is optional, so the current example is correct.

And it appears that the man page has an error. In my testing, it seems like you need the `--notes=foo` syntax to get it to filter for `foo`. Otherwise `foo` gets interpreted as a positional parameter (which leads to a "No such file or directory" error).

Reply:
@cyqsimon can you make a suggestion?

Reply:
What about we add an `--all` example, then change this `--notes` example to show the optional filter usage?

 ```md
- Display all extractable information contained in the ELF file:

`eu-readelf --all {{path/to/file}}`

- Display the contents of all NOTE segments/sections, or of a particular segment/section:

`eu-readelf --notes[={{.note.ABI-tag}}] {{path/to/file}}`
```

Reply:
I'm fine with anything. I just wanted to make an initial page so that it's a springboard for future contributions.

Reply:
What is the bracket notation for btw?

Reply:
> What is the bracket [ ] notation for btw?

It's commonly understood as meaning "optional".  Although not officially documented in contributing guidelines I don't think), there are several examples in TLDR repository:

https://github.com/tldr-pages/tldr/blob/d5dd89e065d6403a901a28c5fe189d9fa396ceea/pages/common/git-notes.md?plain=1#L10-L16

https://github.com/tldr-pages/tldr/blob/d5dd89e065d6403a901a28c5fe189d9fa396ceea/pages/common/svn.md?plain=1#L18-L20

https://github.com/tldr-pages/tldr/blob/d5dd89e065d6403a901a28c5fe189d9fa396ceea/pages/common/tar.md?plain=1#L19-L25

Reply:
Thanks for highlighting the places this is used, we support usage like this to highlight file extensions, etc inside a placeholder i.e. `{{path/to/source.tar[.gz|.bz2|.xz]}}` but for other places like this [{{PATH}}] I would suggest adding it to the description instead (and removing it) as there are a few tldr clients which allows passing an example directly on the host by filling in the placeholder arguments (Fun fact: some of the tldr pages tutorials on YouTube suggests these kinds of clients, but I am not familiar with them, nor sure how popular they are).

Reply:
FYI this search gives a rough filter for such usages:

```bash
rg '^`.+\[[^ ]+\].+`$'
```

Note that this also picks up things like IPv6 and JQ filters, so a manual double-check is necessary.

Reply:
For the sake of clarity, I think we should get rid of the [ ] notation.

Reply:
Can I have opinions on removing [ ] notation alltogether from the program? In my opinion it makes the commands less copypasteable as you'll need to remove the brackets anyway. I don't think integrating them into the program by changing the font inside them similarly to {{ }} is an option either. This would cause commands like the last command in https://github.com/tldr-pages/tldr/blob/main/pages/common/select.md to display wrong info.
There's also a problem similar to man pages. The SYNOPSIS section and its notation is a complete mystery to any newcomer until they start looking it up. I assume we want to avoid similar situations.

Reply:
Agreed, I will try to document that this behavior isn't supported and optional parameters must be specified in the description instead, this weekend.

Reply:
> Agreed, I will try to document that this behavior isn't supported and optional parameters must be specified in the description instead, this weekend.

Should this PR wait for that? Or should we merge now?